vdj_ann/src/transcript.rs

```diff
@@ -7,187 +7,252 @@ use crate::refx::RefData;
-                                stop_codon = true;
-                            }
-                        }
-                        if !stop_codon {
-                            if pass == 1 {
-                                full = true;
-                            }
-                            if logme {
-                                fwriteln!(
-                                    log,
-                                    "{}full length transcript of length {}",
-                                    msg,
-                                    b.len()
-                                );
-                            }
-                        } else if logme {
-                            fwriteln!(
-                                log,
-                                "{}full length stopped transcript of length {}",
-                                msg,
-                                b.len()
-                            );
-                        }
-                    }
-                }
-            }
+    }
+
+    fn order_by(&self) -> u16 {
+        match (
+            self.full_length,
+            self.has_v_start,
+            self.in_frame,
+            self.no_premature_stop,
+            self.has_cdr3,
+            self.has_expected_size,
+            self.correct_ann_order,
+        ) {
+            (Some(_), Some(_), Some(_), Some(_), Some(_), Some(_), Some(_)) => 0,
+            (Some(_), Some(_), Some(_), Some(_), Some(_), Some(_), _) => 1,
+            (Some(_), Some(_), Some(_), Some(_), Some(_), _, _) => 2,
+            (Some(_), Some(_), Some(_), Some(_), _, _, _) => 3,
+            (Some(_), Some(_), Some(_), _, _, _, _) => 4,
+            (Some(_), Some(_), _, _, _, _, _) => 5,
+            (Some(_), _, _, _, _, _, _) => 6,
+            _ => 7,
         }
-        let mut cdr3 = Vec::<(usize, Vec<u8>, usize, usize)>::new();
-        get_cdr3_using_ann(b, refdata, ann, &mut cdr3);
-        if cdr3.is_empty() {
-            if logme {
-                fwriteln!(log, "did not find CDR3");
-            }
-            return false;
+    }
+}
+
+pub struct VdjChainSpecificVJGenes {
+    v_type: String,
+    j_type: String,
+}
+
+impl From<VdjChain> for VdjChainSpecificVJGenes {
+    fn from(chain: VdjChain) -> Self {
+        VdjChainSpecificVJGenes {
+            v_type: format!("{chain}V"),
+            j_type: format!("{chain}J"),
         }
-        let mut too_large = false;
-        const MIN_DELTA: i32 = -25;
-        const MIN_DELTA_IGH: i32 = -55;
-        const MAX_DELTA: i32 = 35;
-        if first_vstart >= 0 && last_jstop >= 0 {
-            let delta = (last_jstop_len + first_vstart_len + 3 * cdr3[0].1.len() as i32 - 20)
-                - (last_jstop - first_vstart);
-            if logme {
-                fwriteln!(log, "VJ delta = {}", delta);
-            }
-            let mut min_delta = MIN_DELTA;
-            if igh {
-                min_delta = MIN_DELTA_IGH;
-            }
+    }
+}
```

Comment:
I don't think we actually need a type here - we only use it once, and we don't ever pass it around as a structure.  I think you can delete this and replace it with these format calls in-place.